library/core/src/ptr/non_null.rs

```diff
@@ -334,6 +334,8 @@ impl<T: ?Sized> NonNull<T> {
     #[rustc_never_returns_null_ptr]
     #[must_use]
     #[inline(always)]
+    //Ensures address of resulting pointer is same as original
+    #[ensures(|result: &*mut T| *result as *const () as usize == self.pointer as *const () as usize)]
```

Comment:
Can this be simplified to `*result == self.pointer`?

Reply:
I tried to use this simplified version of ensures clause: but this error occured`error[E0606]: casting `*mut T` as `usize` is invalid
   --> /Users/dhvanikapadia/verify-rust-std/library/core/src/ptr/non_null.rs:338:33
    |
338 |     #[ensures(|result: &*mut T| *result as usize == self.pointer as usize)]
    |                                 ^^^^^^^^^^^^^^^^
    |
    = help: cast through a thin pointer first`

Reply:
I meant without the cast to `usize`:
```rust
    #[ensures(|result: &*mut T| *result == self.pointer as *mut T)]
```